### ISSUE TITLE
decode方法和论文方法的差异

### ISSUE THREAD

**Issue by lqian** [ouyanghuiyu/centernet_mobilenetv2_ncnn#7]
大神，论文里用maxpool2求极值计算目标中心点的，本项目的genIds函数里判断超过阈值的score输出“候选”目标，然后再NMS，有研究过这两者之间的差异吗?

def _nms(heat, kernel=3):
    pad = (kernel - 1) // 2
    hmax = nn.functional.max_pool2d( heat, (kernel, kernel), stride=1, padding=pad) 
    keep = (hmax == heat).float()
    return heat * keep

**Comment by immarioo** [comment]
我发现预处理过程和centernet也不同，pytorch版本是进行仿射变换resize到512*512，ncnn作者实现是直接根据计算的ratio来进行resize。我在ncnn前向预处理过程中按照pytorch的方式进行预处理发现检测的框都要在下面，对应不上。

**Comment by lqian** [comment]
已经部分移植centernet bbox、keypoints功能，keypoints的finetune部分还没有移植完，不懂python若的货

**Comment by Jumponthemoon** [comment]
> 我发现预处理过程和centernet也不同，pytorch版本是进行仿射变换resize到512*512，ncnn作者实现是直接根据计算的ratio来进行resize。我在ncnn前向预处理过程中按照pytorch的方式进行预处理发现检测的框都要在下面，对应不上。

请问后来是怎么处理的呀。我现在结果也是检测框类别正确，但是位置和size不对。
